front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -203,13 +197,21 @@ const ImportTrainScheduleConfig = ({
         const formattedArrivalTime = `${startDate} ${arrivalTime}`;
         const formattedDepartureTime = `${startDate} ${departureTime}`;
 
+        let stopFor = 0;
+        if (arrivalTime && departureTime) {
+          const arrivalDate = new Date(`${startDate}T${arrivalTime}`);
+          const departureDate = new Date(`${startDate}T${departureTime}`);
+          stopFor = Math.round((departureDate.getTime() - arrivalDate.getTime()) / 1000);
+        }
+
         return {
           id: nextId(),
           uic,
           chCode,
           name: ocpRef,
-          arrivalTime: cleanTimeFormat(formattedArrivalTime),
-          departureTime: cleanTimeFormat(formattedDepartureTime),
+          arrivalTime: formattedArrivalTime,
+          departureTime: formattedDepartureTime,
+          duration: stopFor > 0 ? stopFor : undefined,
```

Comment:
If the user has set the departure time equal to the arrival time, they've requested a zero stop duration. This is semantically different from an undefined stop duration: undefined stop duration means that the train goes through a waypoint without stopping, zero stop duration means that the train stops at a waypoint and immediately restarts.

To handle this, instead of checking whether `stopFor > 0`, we could probably define `stopFor` as a `number | undefined`?

```ts
let stopFor: number | undefined;
```